lib/rules/forbid-dom-props.js

```diff
@@ -66,13 +73,20 @@ module.exports = {
     const forbid = new Map((configuration.forbid || DEFAULTS).map((value) => {
       const propName = typeof value === 'string' ? value : value.propName;
       const options = {
+        disallowList: typeof value === 'string' ? [] : (value.disallowedFor || []),
         message: typeof value === 'string' ? null : value.message,
       };
       return [propName, options];
     }));
 
-    function isForbidden(prop) {
-      return forbid.has(prop);
+    function isForbidden(prop, tagName) {
+      const options = forbid.get(prop);
+      const disallowList = options ? options.disallowList : undefined;
+      return typeof disallowList !== 'undefined'
+        && (typeof tagName === 'undefined'
+        || disallowList.length === 0
+        || (disallowList.length > 0
+        && disallowList.indexOf(tagName) !== -1));
```

Comment:
```suggestion
      return typeof disallowList !== 'undefined' && (
        typeof tagName === 'undefined'
        || disallowList.indexOf(tagName) !== -1
      );
```
indexOf is free if the array's empty.

Reply:
ahh right, this is "is forbidden". what about this?
```suggestion
      const disallowList = options ? options.disallowList : [];
      return typeof tagName !== 'undefined' && disallowList.indexOf(tagName) !== -1;
```
we might not even need the tagName check

Reply:
ahhh sorry, i get it. When the disallow list is empty, then everything is forbidden.

ok - so let's try a different approach. (I don't think the tagName should ever be undefined?) Something like the following:
Currently, there's a `forbid` map of prop name to options. I think we should update that code so that instead of it having `options.disallowList`, it has an `options.isAllowed` function.

That function then would be:
```js
(tag) => options.disallowList && options.disallowList.length > 0 ? options.disallowList.indexOf(tag) === -1 : true
```

Would that work?

Reply:
It would contain a predicate function.

Reply:
Neither? Predicate functions are great. What i was thinking is that the confusing logic could be wrapped up in a function when creating the Map, so that the individual check here could be simpler.